feat: add integration with snacks picker

iniw opened the pull request:
This also drive-by fixes #703, which I can put in a separate PR if that'd be preferred.

mikavilpas commented:
Cool, thanks! I'll try to test this later today.

The idea for #703 and making the grep integrations a bit nicer is great too. I added the changes related to the types and will merge that in separately in https://github.com/mikavilpas/yazi.nvim/pull/705

iniw commented:
> The idea for #703 and making the grep integrations a bit nicer is great too. I added the changes related to the types and will merge that in separately in #705.

Ah yes, sorry, I completely forgot about docs and type definitions and stuff like that. I have gone ahead and added documentation for the snacks picker and removed the #703 fix.

iniw commented:
Is it fine for me to not write the js unit tests? I didn't really figure out the workflow for those... and the snacks picker is currently in extremely active development, maybe it'd be easy for those tests to break.

mikavilpas commented:
It's fine! It's actually a good idea.

I have had one or two tests for the grug far integration that try to open it but don't check anything. I might add one of those myself.
